### ISSUE TITLE
localized file names issue

### ISSUE THREAD

**Issue by a2nt** [silverstripe/silverstripe-framework#6384]
If I will upload a file with localized name it will have wrong file name for example:
"ûüù.jpg" will be saved as "jpg" which will make file extension undetectable.

**Comment by tractorcow** [comment]
I assume you discovered this in 3.x? 4.x will have a similar, yet different problem. :)

**Comment by a2nt** [comment]
yes, that's right. I think file name shall use file ID or some random name in case after removing localized chars file name is empty or transliterated it's security breach right now.
